install/helm/agones/templates/crds/_gameserverspecvalidation.yaml

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# validation for a gameserver spec
+
+{{- define "gameserver.validation" }}
```

Comment:
@Kuqd @fooock - see any issues with me shifting this validation into a template like this?

Seems like the correct way to do it, but wanted to double check in case you could see anything obviously bad?

Reply:
For me it's ok. Small things:

* I would add a small documentation above the `define`, line `{{ /*  Validation for a gameserver spec .... */ }}`
* Why `{{ end - }}` instead of `{{ - end }}` ??
* Normally these templates are inside the `_helpers.tpl` file, but by organization, I would leave it like that.

Reply:
Good call on the first two. Done!.

Re: 
> Normally these templates are inside the _helpers.tpl file, but by organization, I would leave it like that.

So I looked at https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/templates.md

> Template files should have the extension .yaml if they produce YAML output. The extension .tpl may be used for template files that produce no formatted content.

So I made it a .yaml file with an underscore, since it's a pretty big yaml block. WDYT?